src/pages/workspace/WorkspaceMoreFeaturesPage.tsx

```diff
@@ -86,6 +101,13 @@ function WorkspaceMoreFeaturesPage({policy, route}: WorkspaceMoreFeaturesPagePro
             disabled: hasAccountingConnection,
             pendingAction: policy?.pendingFields?.areCategoriesEnabled,
             action: (isEnabled: boolean) => {
+                if (hasAccountingConnection) {
+                    setConnectionWarningModalState({
+                        isOpen: true,
+                        itemType: 'organize',
+                    });
+                    return;
+                }
```

Comment:
We repeated this piece of code few times, I think we should reuse it with an `useCallback`?

Reply:
I am not sure if it is necessary, we're not relying on the previous state that we really need to put it inside useCallback. The `hasAccountingConnection` is also dependent on the prop.

And about repetition, it's a setState call. I could wrap it inside the method but looks like an overkill.

Reply:
I think we should, including ` Policy.enablePolicyCategories(policy?.id ?? '', isEnabled);` line below.
Also can you complete the checklist please 😄 ?

Reply:
Working through the checklist.

Reply:
With enablePolicyCategories it might make sense. Let me check.

Reply:
@lakchote Do you think it makes sense to move this code inside a useCallback?

Reply:
I do agree with @mananjadhav, I don't think there's a benefit _here_ to use `useCallback()`. It's not a computational intensive operation, we're just setting state. It can be tempting to use it to avoid repetition, but using these memorized functions come at a cost on intial render. Given the fact it's just setting state, I'd say it's a premature optimization scenario and we can keep it that way.

Happy to be proven wrong though, if you can give me arguments proving me otherwise @hungvu193 😄

Reply:
I think that's fine though. We can keep it this way :)